packages/amplify-graphql-auth-transformer/src/__tests__/non-model-auth.test.ts

```diff
@@ -173,4 +182,68 @@ describe('@auth directive without @model', () => {
     expectNoDirectives(queryType!);
     expectOneDirective(getField(queryType, 'getSecret'), 'aws_iam');
   });
+
+  test('generates the right IAM policy for AuthRole using private auth', () => {
+    const validSchema = `
+      type Query {
+          getSecret: String!
+          @auth (
+            rules: [
+              { allow: private, provider: iam },
+            ])
+          @function (name: "getSecret-\${env}")
+      }
+      `;
+
+    const testTransformParams = {
+      schema: validSchema,
+      authConfig,
+      transformers: [new ModelTransformer(), new AuthTransformer(), new FunctionTransformer()],
+    };
+
+    const out = testTransform(testTransformParams);
+    expect(out).toBeDefined();
+    const resources = out.rootStack.Resources;
+    expect(resources).toBeDefined();
+
+    const authRolePolicy = getObjectWithKeyPrefix('AuthRolePolicy', resources!);
+    const unauthRolePolicy = getObjectWithKeyPrefix('UnauthRolePolicy', resources!);
+    expect(authRolePolicy).toBeDefined();
+    expect(unauthRolePolicy).toBeUndefined();
+    expect(authRolePolicy?.Properties.PolicyDocument.Statement[0].Resource).toBeDefined();
+    expect(authRolePolicy?.Properties.PolicyDocument.Statement[0].Resource).not.toEqual({});
+  });
+
+  test('generates the right IAM policies for AuthRole and UnauthRole using public auth', () => {
+    const validSchema = `
+      type Query {
+          getSecret: String!
+          @auth (
+            rules: [
+              { allow: public, provider: iam },
+            ])
+          @function (name: "getSecret-\${env}")
+      }
+      `;
+
+    const testTransformParams = {
+      schema: validSchema,
+      authConfig,
+      transformers: [new ModelTransformer(), new AuthTransformer(), new FunctionTransformer()],
+    };
+
+    const out = testTransform(testTransformParams);
+    expect(out).toBeDefined();
+    const resources = out.rootStack.Resources;
+    expect(resources).toBeDefined();
+
+    const authRolePolicy = getObjectWithKeyPrefix('AuthRolePolicy', resources!);
+    const unauthRolePolicy = getObjectWithKeyPrefix('UnauthRolePolicy', resources!);
+    expect(authRolePolicy).toBeDefined();
```

Comment:
The [docs](https://docs.amplify.aws/react/build-a-backend/graphqlapi/customize-authorization-rules/#authorization-strategies) seem to indicate that `public` auth allows access to `unauthenticated users/devices` contrary to the test here where both authenticated and unauthenticated entities are given access.

Reply:
This tests asserts the policies are generated, not what permissions are being assigned. We can of course write a test to assert the correct permissions behavior, but that's a different story.